### ISSUE TITLE
Unable to login from SLURM frontend node to compute nodes with CentOS 

### ISSUE THREAD

**Issue by verdurin** [gc3-uzh-ch/elasticluster#328]
On a newly-created `CentOS` `SLURM` cluster, I'm unable to SSH from the frontend node to the compute nodes. This is possible on `Ubuntu` clusters.

**Comment by cocoy** [comment]
Hmm, is this using AWS? I'm sure the ssh ports are open and the username is root for CentOS.
Do you have the error logs? @verdurin

**Comment by riccardomurri** [comment]
I cannot reproduce this on the local cloud, using OpenStack and a CentOS 7.2.1511 image::

    (elasticluster-dev)➜  elasticluster git:(master) elasticluster ssh slurm-on-centos
    ...

    [centos@master001 ~]$ ssh worker001
    Last login: Mon Oct 24 13:36:04 2016 from ...
    [centos@worker001 ~]$ hostname
    worker001
    [centos@worker001 ~]$ cat /etc/redhat-release
    CentOS Linux release 7.2.1511 (Core) 
    [centos@worker001 ~]$ logout
    Connection to worker001 closed.
    
    [centos@master001 ~]$ ssh worker002
    Last login: Mon Oct 24 13:36:04 2016 from ...
    [centos@worker002 ~]$ hostname
    worker002

**Comment by verdurin** [comment]
On Thu, Oct 20, 2016 at 12:24:13AM -0700, Rodney Quillo wrote:
> Hmm, is this using AWS? I'm sure the ssh ports are open and the username is root for CentOS.
> Do you have the error logs? @verdurin 
> 

Yes, should have provided the full context.

This is on OpenStack.

I'll send the full logs later.

**Comment by verdurin** [comment]
Here you go, with a brand new cluster:

```
elasticluster ssh ctslurm11
Last login: Tue Nov  1 21:30:33 2016 from 
[centos@frontend001 ~]$ ssh compute001
Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
[centos@frontend001 ~]$ 
```

**Comment by verdurin** [comment]
Hmm. Just changed the logging level for `sshd` on the compute node, and then logging in from the head node worked... 

Possibly the result of reload `sshd`?

**Comment by verdurin** [comment]
Yes, confirmed by creating a new cluster.

The first attempt to SSH from the head node to a compute node fails.

After connecting from outside the cluster, then restarting `sshd` on the compute node, SSH from the head node works.

**Comment by verdurin** [comment]
Further confirmation is that the same phenomenon occurs with SSH from `compute001` to `compute002` i.e. it only works after restarting `sshd` on `compute002`.

**Comment by riccardomurri** [comment]
@verdurin is this issue still current?

**Comment by verdurin** [comment]
@riccardomurri I haven't had the chance to retest with the latest code. Will try to do so this week.

**Comment by hackermd** [comment]
I had the same problem. Restarting `sshd` service on compute nodes also solved it for me. It seems that updates of the configurations in `sshd_config` with respect to host-based authentication are not applied. The culprit might be Ansible's `notify`. I've included a manual `systemctl restart sshd.service` at the end of my play and that did the trick for me.

Looking at the Ansible output, I realized that the handler was not triggered:

```
RUNNING HANDLER [common : restart sshd] ********************************************************************
skipping: [compute001] => {"changed": false, "skip_reason": "Conditional result was False", "skipped": true}
```
